player.cpp

```diff
@@ -0,0 +1,40 @@
+#include "player.h"
+#include <string>
+#include <cmath>
+
+Player::Player(const std::string& name, int health, int damage) : Character(name, health, damage) { xp = 0; level = 1, maxHP = health; }
+
+int Player::GetLevel() const
+{
+	return level;
+}
+
+void Player::Attack(Player& enemy) {
```

Comment:
Az ősosztálynak is van `Attack` metódusa, ennek azt kellene override-olnia. És nem látom okát, hogy csak másik `Player`t tudjon támadni, `Character`t ne.